flexdll.c

```diff
@@ -47,6 +47,8 @@ typedef struct dlunit {
 } dlunit;
 typedef void *resolver(void*, const char*);
 
+static volatile HANDLE units_mutex = INVALID_HANDLE_VALUE;
```

Comment:
Is the `volatile` really needed?

Reply:
According to [this page](https://learn.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedcompareexchangepointer) the first destination parameter to `InterlockedCompareExchangePointer` should be `volatile`

Reply:
Could it have to do with parallel access, e.g., in the presence of races to initialize it?

Reply:
OK, after looking more into this I'm convinced (I'm no memory-model expert!).
What's important to ensure synchronization is the atomic update and memory fence.
Addressed in 47480bc